README.md

```diff
@@ -717,6 +778,64 @@ postgres:
+
+The **LDAP_XXX_SEARCH_BASE** is the search base dn. It will default to **LDAP_DEFAULT_SEARCH_BASE** as **LDAP_XXX_SEARCH_SCOPE** will default to **LDAP_DEFAULT_SEARCH_SCOPE** (which defaults to *sub*).
+
+The **LDAP_XXX_FILTER** must be a valid LDAP query filter. For a documentation of LDAP query filters you can look at https://ldap.com/ldap-filters/. For a list of valid replacement tokens please look at http://www.postfix.org/ldap_table.5.html in the section *query_filter*. Some examples:
+
+```
+LDAP_DOMAIN_FILTER="(&(mail=*@%s)(objectClass=mailAccount))"
+
+LDAP_MAILBOX_FILTER="(&(mail=%s)(objectClass=mailAccount))"
+
+LDAP_SENDER_FILTER="(&(|(mail=%s)(mailalias=%s))(objectClass=mailAccount))"
+```
+
+The **LDAP_XXX_ATTRIBUTE** specifies which attribute of the found LDAP objects will be used. Usually these are either *mail*, *uid*, *mailalias* or *mailacceptinggeneralid* but may be completly different ones depending on your LDAP setup.
+
+
+**LDAP_XXX_FORMAT** can be used to reformat the result. E.g. you can use `LDAP_MAILBOX_FORMAT="/var/mail/vhosts/%d/%s/mail/"` to set a fixed path for the mailbox location if the path is not stored within LDAP.
+
+The optional **FORWARD** and **GROUP** lookups are technically identical to the **ALIAS** lookup and could be used interchangeably but are intended for additional alias/group/forward lookups. So you can use aliases using an alias field in your user objects. Forwards as source and destination mapping fields in forwarding objects and group address and group member emails in group objects. But you can also use them in different ways to suit your system.
+
+Then you also have to provide the lookups for dovecot. These will probably be similar to your postfix lookups but may and will differ in some cases. The variables neccessary are **LDAP_DOVECOT_USER_ATTRS**, **LDAP_DOVECOT_USER_FILTER**, **LDAP_DOVECOT_PASS_ATTRS**, **LDAP_DOVECOT_PASS_FILTER**, **LDAP_DOVECOT_ITERATE_ATTRS**, **LDAP_DOVECOT_ITERATE_FILTER**. They correspond directly to the dovecot variables of the same name. While the user and pass attributes and filters are required, the iterate attributes and filters are not. For more detailed information please look at https://wiki.dovecot.org/AuthDatabase/LDAP/Userdb. Note that multiple attribures may be required per query and must be provided in a different form than for postfix! Here are some examples:
+
+```
+LDAP_DOVECOT_USER_ATTRS="=home=/var/mail/vhosts/%d/%n/,=mail=maildir:/var/mail/vhosts/%d/%n/mail/,mailuserquota=quota=quota_rule=*:bytes=%\$$"
```

Comment:
If you use the `\$$` in your compose file it will not work. Just `%$` in the compose file is enough.

Reply:
Actually, here it needs to be `%\$$` to work. $$ is neccessary for docker compose and will result in a single $ (https://docs.docker.com/compose/compose-file/#variable-substitution). Not sure where the \ ist removed, might be the build script for the config files?

%$ is what needs to be in the config file in the end. But to get there, you sadly need those two escapes, at least that is how I got it to work (and only this way).

Reply:
Reading the information your right. I forgot to mention that on kubernetes I had to write it as I wrote my comment.

Reply:
Good to know. I will add this to the documentation with more details on the escaping (because it took me quite some time to figure it out aswell).

Reply:
I think \ is possibly removed by the shell since it is originally specified on the command line in Makefile.